File: algorithm/stack3.py
# https://www.acmicpc.net/problem/2812
# 골드3

n, k=map(int, input().split())
number=list(input())
# number=list(map(int, input()))

answer=[]
cnt=k
for num in number:
    while answer and cnt > 0 and answer[-1] < num:
        answer.pop();
        cnt-=1
    answer.append(num)
print(''.join(answer[:n-k]))

File: algorithm-js/stack3.js
// 골드3 https://www.acmicpc.net/problem/2812
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, K] = input[0].split(' ');
const inputNumbers = input[1].split('');

const stack = [];
let count = K;

inputNumbers.forEach((number) => {
    while (count > 0 && number > stack[stack.length - 1]) {
        stack.pop();
        count--;
    }
    stack.push(number);
});

console.log(stack.join(''));